### ISSUE TITLE
Wrong error message for $Shape

### ISSUE THREAD

**Issue by taylorgoolsby** [facebook/flow#7044]
https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgCqAzngE4BCAhqWALxgDeqYrb7HrArqWQJIATAFxhiGMgEsAdgHMANC05K2UqgFs8IsZNmLlSgL7oAxnCliwGPGICMIgCQBlABZV8AHhLlqpAHz0mRR5yQREAchswhVZVDXCAJijFKzFwgGYkoyA

Minor bug. Just noticed that the error message is backwards. 

Expected:
```
Cannot assign object literal to `test1` because property `test` exists in object literal [1] but is missing in `UserBase` [2].
```

Got:
```
Cannot assign object literal to `test1` because property `test` is missing in object literal [1] but exists in `UserBase` [2].
```

**Comment by talbenari1** [comment]
Just encountered this as well. Simpler repro:
```js
({ x: 1 }: $Shape<{}>)
```

**Comment by fdc-viktor-luft** [comment]
$Shape seems to be totally broken or do I get anything wrong?
flow --version --> 0.86.0

https://flow.org/try/#0FAFwngDgpgBAKlAziGBeGBvGBDAXDZAJwEsA7AcwBoYAjfUgVwFsapCYBfAbmGAGMA9qWS1shfABIAygAts0ADwJkAPjSYc+AOQyBWzj2AAKGmPzKQASh4mxAOjoxGLNteBA

**Comment by dsainati1** [comment]
Should be fixed in an upcoming release.

**Comment by jamesisaac** [comment]
From the Try Flow seems like this was fixed in v0.90
